perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,100 @@
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_virt_elixirBonaFideStatus.class);
+
+	private final Pattern addUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] added to User:\\[(.*)]");
+	private final Pattern removeUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] removed from User:\\[(.*)]");
+	private final Pattern extSourceRems = Pattern.compile("cz.metacentrum.perun.core.impl.ExtSourceREMS");
+	private final Pattern affiliationsAttribute = Pattern.compile("Attribute:\\[(.*)friendlyName=<eduPersonScopedAffiliations>(.*)] set for User:\\[(.*)]");
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		Attribute attribute = new Attribute(attributeDefinition);
+		String url = "http://www.ga4gh.org/beacon/bonafide/ver1.0";
+
+		try {
+			Attribute statusAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF + ":elixirBonaFideStatusREMS");
+			String attrDefValue = (String) statusAttr.getValue();
+			if (attrDefValue != null && !attrDefValue.isEmpty()) {
+				attribute.setValue(attrDefValue);
+			} else {
+				Attribute uesAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":eduPersonScopedAffiliations");
+				if (uesAttribute.getValue() != null) {
+					List<String> affiliations = (List<String>) uesAttribute.getValue();
+					for (String affiliation : affiliations) {
+						if (affiliation != null && affiliation.startsWith("faculty@")) {
+							attribute.setValue(url);
+							break;
+						}
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read elixirBonaFideStatusREMS from user " + user.getId(), e);
+		}
+		return attribute;
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl sess, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<String> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
+
+		Matcher addUserExtSourceMatcher = addUserExtSource.matcher(message);
+		Matcher removeUserExtSourceMatcher = removeUserExtSource.matcher(message);
+		Matcher extSourceRemsMatcher = extSourceRems.matcher(message);
+		Matcher affiliationsAttributeMatcher = affiliationsAttribute.matcher(message);
+
+		User user;
+		Attribute attrVirtRemsLogins;
+
+		if (affiliationsAttributeMatcher.find() || (
+				extSourceRemsMatcher.find() && (addUserExtSourceMatcher.find() || removeUserExtSourceMatcher.find()))) {
+
+			user = (new ModulesUtilsBlImpl()).getUserFromMessage(sess, message);
```

Comment:
Yes, it is. This method is not implemented in the interface, and in class ModulesUtilsBlImpl it is not marked as a static method. And actually, I think there is a bug inside of it. On line no. 1011 there is 
`user = (User) user` which probably should have been `user = (User) perunBean`.

Reply:
You are right, it is not static method but it is implemented in ModulesUtilsBl on the line 428 as far as I can say from master branch.
And yes, thank you, there is a bug. I will create hot fix for it.

Reply:
This shoudl be the right way how to call it: perunSession.getPerunBl().getModulesUtilsBl().getUserFromMessage(perunSession, message);

Reply:
@stavamichal Thanks, I have found out it already yesterday, I guess that editing comments here is not a good idea since it probably does not let you know about editing the comment.

Reply:
@Vojtech-Sassmann yes, I did not get info about changing comment. We already merged hot fix for mentioned bug to the production branch. Thank you.